### PR TITLE
DOC: add doc strings to FigureCanvasAgg methods

### LINKED ISSUE #6980: problem accessing canvas on MacOS 10.11.6 with matplotlib 2.0.0b3
When I try running the below code directly with python (or within IPython 5.1.0 after running `%matplotlib qt4` with the `matplotlib.use('Qt4Agg')` line removed), 

```python
import matplotlib
matplotlib.use('Qt4Agg')
import matplotlib.pyplot as plt
import numpy as np

f = plt.gcf()
plt.imshow(np.random.rand(100, 100))
plt.draw()
im = np.fromstring(f.canvas.tostring_rgb(), dtype=np.uint8)
```
I observe the following error:

```
Traceback (most recent call last):
  File "test_renderer.py", line 9, in <module>
    im = np.fromstring(f.canvas.tostring_rgb(), dtype=np.uint8)
  File "/Users/lebedov/anaconda2/lib/python2.7/site-packages/matplotlib/backends/backend_agg.py", line 484, in tostring_rgb
    return self.renderer.tostring_rgb()
AttributeError: 'FigureCanvasQTAgg' object has no attribute 'renderer'
```

If I'm in IPython and run the script a second time after the above exception, no error occurs. Moreover, if I add `f.canvas.draw()` before I try to run `f.canvas.tostring_rgb()`, the error goes away. Also, the error appears to occur with backends other than qt4 such as tk (although not with agg).

Could this be related to #5058?

I'm running matplotlib 2.0.0b3 with Python 2.7.12 and PyQt4 4.11.4 (all installed in Anaconda from conda-forge) on MacOS 10.11.6.

### PR DESCRIPTION
**Pull request by tacaswell**
closes #6980

### CODE REVIEW

File: lib/matplotlib/backends/backend_agg.py [hunk 1]
```diff
@@ -479,16 +479,44 @@ def get_renderer(self, cleared=False):
         return self.renderer
 
     def tostring_rgb(self):
+        '''Get the image as an RGB byte string
+
+        `draw` much be called at least once before this function will work and
```
Comment (QuLogic) [inline lib/matplotlib/backends/backend_agg.py]: s/much/must/

Comment (QuLogic) [inline lib/matplotlib/backends/backend_agg.py]: And repeat in every case.

Comment (tacaswell) [inline lib/matplotlib/backends/backend_agg.py]: Sign I should go to bed....

Comment (tacaswell) [inline lib/matplotlib/backends/backend_agg.py]: Fixed an force pushed

**Pull request merged**